File: Capitulo-01/Pesquisa-Binária/Códigos/Java/pesqisa-binaria/src/PesquisaBinaria.java
public class PesquisaBinaria {
    public static Integer pesquisaBinaria(int[] lista, int alvo) {
        int baixo = 0;
        int alto = lista.length - 1;

        while (baixo <= alto) {
            int meio = (baixo + alto) / 2;

            if (lista[meio] == alvo) {
                return meio;
            } else if (lista[meio] < alvo) {
                baixo = meio + 1;
            } else {
                alto = meio - 1;
            }
        }

        return null;
    }

    public static void main(String[] args) {
        int[] minhaLista = { 1, 3, 5, 7, 9 };

        System.out.println(pesquisaBinaria(minhaLista, 1));
        System.out.println(pesquisaBinaria(minhaLista, 3));
        System.out.println(pesquisaBinaria(minhaLista, 5));
        System.out.println(pesquisaBinaria(minhaLista, 7));
        System.out.println(pesquisaBinaria(minhaLista, 9));
        System.out.println(pesquisaBinaria(minhaLista, 20));
    }
}

File: Capitulo-01/Pesquisa-Binária/Códigos/Ruby/pesquisa_binaria.rb
def busca_binaria(lista, alvo)
  baixo = 0
  alto = lista.length - 1

  while baixo <= alto
    meio = (baixo + alto) / 2

    if lista[meio] == alvo
      return meio
    elsif lista[meio] < alvo
      baixo = meio + 1
    else
      alto = meio - 1
    end
  end

  nil
end

minha_lista = [1, 3, 5, 7, 9]

puts busca_binaria(minha_lista, 1)
puts busca_binaria(minha_lista, 3)
puts busca_binaria(minha_lista, 5)
puts busca_binaria(minha_lista, 7)
puts busca_binaria(minha_lista, 9)
puts busca_binaria(minha_lista, 20)
